Complete group feeds functionality

a-j-sheilla opened the pull request:
- Add group join request functionality for users to request joining groups
- Add group creator ability to accept/reject join requests
- Only group creators can manage join requests (authorization enforced)
- Create frontend interface for joining groups and managing requests

 1. Group Invitations
  - Group creator AND all active members can invite other users
  - Invited users receive invitations they can accept or reject
  - Dedicated /groups/invites page to view pending invitations
  - Accept/Reject invitation endpoints with proper status updates
  - Users excluded from invitation list if already members

2. Join Requests

  -  Users can browse all groups and request to join
  - Only group creators can view, accept, or reject join requests
  - Dedicated /groups/join-requests page for creators to manage requests
  - Pending request status displayed on groups page
  - Proper filtering to show only groups created by current user

  3. Group Events

  - Any group member can create events
  - Events include: Title, Description, Date/Time
  - Two response options: "Going" and "Not Going"
  - Real-time response counts displayed
  - Users can change their response at any time
  - Dedicated /groups/:id/events page with event creation modal

Commit:
Implement POST /groups/:id/join endpoint with frontend

- Add group join request functionality for users to request joining groups
- Add group creator ability to accept/reject join requests
- Only group creators can manage join requests (authorization enforced)
- Create frontend interface for joining groups and managing requests

Backend changes:
- Add GetPendingJoinRequests and RejectJoinRequest methods to repository
- Add corresponding service layer methods with proper authorization
- Add new handler methods for managing join requests
- Update routes to handle new endpoints
- Add GroupJoinRequest model

Frontend changes:
- Enhanced groups page with join functionality and status display
- New join-requests management page for group creators
- Real-time UI updates when processing requests
- Proper error handling and user feedback

Endpoints implemented:
- POST /groups/:id/join - Request to join group
- POST /groups/:id/join?action=accept - Accept join request (creators only)
- POST /groups/:id/join?action=reject - Reject join request (creators only)
- GET /groups/:id/join-requests - Get pending requests (creators only)

anamivale commented on backend/internal/repository/group_repository.go:
```diff
@@ -154,3 +154,63 @@ func (r *GroupRepository) IsGroupCreator(groupID uint, userID string) (bool, err
 }
+
+// GetPendingJoinRequests retrieves all pending join requests for a group.
+func (r *GroupRepository) GetPendingJoinRequests(groupID uint) ([]model.GroupJoinRequest, error) {
```
Did you define this model?

valeria1234fasi committed:
Merge branch 'main' into group-join-endpoint

valeria1234fasi committed:
fix : fix migration conflicts

valeria1234fasi committed:
feat: add migration files for events and event_responses tables

valeria1234fasi committed:
feat: Implement group events page with event creation and response handling

- Added GroupEventsPage component to display events for a specific group.
- Implemented event fetching, loading, and error handling.
- Added functionality to respond to events (going/not going).
- Created CreateEventModal for event creation with form validation.
- Integrated event time formatting for better display.
- Updated GroupPage to include a link to the events page.
- Added GroupInvitesPage to manage group invitations with accept/reject functionality.
- Implemented JoinRequestsPage to handle join requests for groups created by the user.
- Enhanced UI with consistent styling and improved user experience.

valeria1234fasi committed:
feat: Add event management functionality with creation, retrieval, response handling, and deletion

valeria1234fasi committed:
feat: Add event handling routes for creating, retrieving, and responding to events

valeria1234fasi committed:
feat: Implement group invitation handling with accept, reject, and retrieval functionalities

valeria1234fasi committed:
feat: Add endpoint to retrieve available users for group invitations

valeria1234fasi committed:
feat: Update user invitation modal to fetch available users from group-specific endpoint

anamivale reviewed (approved):
Good work
